bin/code_connect.py

```diff
@@ -68,17 +66,17 @@ def get_code_binary() -> Path:
 
     # Every entry in ~/.vscode-server/bin corresponds to a commit id
     # Pick the most recent one
-    code_repos = sort_by_access_timestamp(Path.home().glob(".vscode-server/bin/*"))
+    code_repos = sort_by_access_timestamp(Path.home().glob(".vscode-server/cli/servers/*"))
```

Comment:
(Sorry, I'm just a random passer-by, not the author)

Maybe we can put a more concrete condition for repo candidates, otherwise:

```
$ code
Traceback (most recent call last):
  File "/home/USER/.code-connect/bin/code_connect.py", line 120, in <module>
    main()
  File "/home/USER/.code-connect/bin/code_connect.py", line 113, in main
    proc = sp.run(args)
           ^^^^^^^^^^^^
  File "/home/USER/.asdf/installs/python/3.11.9/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/USER/.asdf/installs/python/3.11.9/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/USER/.asdf/installs/python/3.11.9/lib/python3.11/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
NotADirectoryError: [Errno 20] Not a directory: '/home/USER/.vscode-server/cli/servers/lru.json/server/bin/remote-cli/code'
```

```
$ l ~/.vscode-server/cli/servers
total 12K
drwxr-xr-x 3 USER USER 4.0K May  6 08:46 Stable-b58957e67ee1e712cebf466b995adf4c5307b2bd
drwxr-xr-x 3 USER USER 4.0K Apr 25 16:32 Stable-e170252f762678dec6ca2cc69aba1570769a5d39
-rw-r--r-- 1 USER USER  101 May  6 08:46 lru.json
```

Reply:
Hey, no apologies necessary. I very much like more robust solutions, and this is certainly the way to go. I'll probably dig into this tomorrow a little.

Reply:
@kissge I rewrote the cli executable search code to be a lot more broad but also check against the output to make sure we're only manipulating the right code binaries. It seems to work, though I don't have access to my non-local system to properly test at the moment, nor do I have multiple versions installed to confirm it selects the most up-to-date executable.

Let me know how it works for you!